### PR TITLE
Add ones_like and zeros_like

### PR DESCRIPTION
**Pull request by alykhantejani**
As part of #1875, this PR adds numpy-style `ones_like` and `zeros_like` to `torch`. Example usage:

```Python
import torch
input = torch.Tensor(2,3)
torch.ones_like(input)

 1  1  1
 1  1  1
[torch.FloatTensor of size 2x3]
```

### CODE REVIEW

**Comment by vadimkantorov** [comment]
should these be supported on Variables too?

**Comment by alykhantejani** [comment]
@vadimkantorov probably. There is a list tracking these missing ops from Variables [here](https://github.com/pytorch/pytorch/issues/440). 

However, I have an [open PR](https://github.com/pytorch/pytorch/pull/2212) adding `zeros_` to Variables but it's unclear if this is still desired, as @apaszke raised some concerns on that PR.

File: torch/lib/TH/generic/THTensorMath.c [hunk 1]
```diff
@@ -1927,6 +1927,18 @@ void THTensor_(zeros)(THTensor *r_, THLongStorage *size)
   THTensor_(zero)(r_);
 }
 
+void THTensor_(zeros_like)(THTensor *r_, THTensor *input)
+{
+  THTensor_(resize)(r_, THTensor_(newSizeOf)(input), NULL);
```
Comment (colesbury) [inline torch/lib/TH/generic/THTensorMath.c]: I think this will leak `THLongStorage*` returned by `newSizeOf`.

Instead:

`THTensor_(resizeAs)(r_, input)`

Comment (alykhantejani) [inline torch/lib/TH/generic/THTensorMath.c]: @colesbury yes, you're right - nice spot. Will change now.

File: torch/lib/TH/generic/THTensorMath.c [hunk 2]
```diff
@@ -1927,6 +1927,18 @@ void THTensor_(zeros)(THTensor *r_, THLongStorage *size)
   THTensor_(zero)(r_);
 }
 
+void THTensor_(zeros_like)(THTensor *r_, THTensor *input)
```
Comment (fmassa) [inline torch/lib/TH/generic/THTensorMath.c]: nit: in TH/THC, almost all functions are camelCase. Should we enforce this here for consistency? (and let snake_case in the python interface)

Comment (alykhantejani) [inline torch/lib/TH/generic/THTensorMath.c]: This is now done

File: torch/lib/TH/generic/THTensorMath.c (hunk 1 above)
Comment (alykhantejani) [inline torch/lib/TH/generic/THTensorMath.c]: I've made the changes here and to THC as well

**Review by apaszke — APPROVED** [review]

**Comment by soumith** [comment]
this is now merged into master. thanks a lot Aly!

**Pull request closed without merging**